common/datavalues/src/types/type_coercion.rs

```diff
@@ -323,7 +322,7 @@ pub fn merge_types(lhs_type: &DataTypePtr, rhs_type: &DataTypePtr) -> Result<Dat
             let b = rhs_type.as_any().downcast_ref::<ArrayType>().unwrap();
```

Comment:
I suggest to remove all `downcast_ref`, since we have `try_from`, `try_into` now!